### PR TITLE
fix(list): remove unnecessary z-index

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7599

## Summary

- removes z-index on a container within the list component

### CODE REVIEW

File: packages/calcite-components/src/components/list/list.scss [hunk 1]
```diff
@@ -13,9 +13,7 @@
   flex
   w-full
   flex-col
-  bg-transparent
-  relative
-  z-default;
+  bg-transparent;
```
Comment (macandcheese) [inline packages/calcite-components/src/components/list/list.scss:16]: We might still need to bump the z-index of the `filter-enabled` UI elements - with this change it is behind the focused list element:

https://github.com/Esri/calcite-design-system/assets/4733155/b57ad989-531a-410b-868d-a20c2b5b1109


https://github.com/Esri/calcite-design-system/assets/4733155/07f6b932-7e9e-4cc8-ba90-e7001f094b53

Comment (driskull) [reply packages/calcite-components/src/components/list/list.scss:16]: Nice catch @macandcheese. I removed the z-index on the active element. Not sure why we had it there. @jcfranco any idea?

Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.scss:16]: I don't recall. Nice cleanup though! 🧼

**Review by jcfranco — APPROVED** [review]
Low-risk approved! 🥇

**Comment by driskull** [comment]
@macandcheese can you test this one again? Por favor 🙇

**Review by macandcheese — APPROVED** [review]
Looks good with this change!

**Pull request merged**